### PR TITLE
Adding testnet and Mock Faucet Tokens

### PR DESCRIPTION
**Pull request by 0xPrimata**
Cleared up dependencies and files from the original strucuture (this is a fork of the explorer)

Issues:

Right now in a hellhole of contexts since we need to expose the faucet to three wallets (aptos, sui, evm) trying to figure out what's happening.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -128,11 +130,16 @@ ReactDOM.render(
...
     >
       <QueryClientProvider client={queryClient}>
+        <QueryClientProvider client={queryClient}>
         <AptosWalletAdapterProvider plugins={wallets} autoConnect={true}>
+        <SuiClientProvider networks={networkConfig} defaultNetwork="devnet">
+        <WalletProvider>
           <BrowserRouter>
-            <ExplorerRoutes />
           </BrowserRouter>
-        </AptosWalletAdapterProvider>
```
Comment (andygolay) [inline src/index.tsx:135]: I updated `index.tsx` to get it to load. You'll need to change the `wagmiConfig` values.

```
import React from "react";
import ReactDOM from "react-dom";
import '@mysten/dapp-kit/dist/index.css';

import {BrowserRouter, Route, Routes} from "react-router-dom";
import LandingPage from "./pages/LandingPage/Index";
import {QueryClient, QueryClientProvider} from "@tanstack/react-query";
import {
  AptosWalletAdapterProvider,
  NetworkName,
} from "@aptos-labs/wallet-adapter-react";
import {PetraWallet} from "petra-plugin-wallet-adapter";
import {PontemWallet} from "@pontem/wallet-adapter-plugin";
import {MartianWallet} from "@martianwallet/aptos-wallet-adapter";
import {RiseWallet} from "@rise-wallet/wallet-adapter";
import {FewchaWallet} from "fewcha-plugin-wallet-adapter";
// import {MSafeWalletAdapter} from "msafe-plugin-wallet-adapter";
import {StatsigProvider} from "statsig-react";
import {BloctoWallet} from "@blocto/aptos-wallet-adapter-plugin";
import {NightlyWallet} from "@nightlylabs/aptos-wallet-adapter-plugin";
import {OpenBlockWallet} from "@openblockhq/aptos-wallet-adapter";
import {TokenPocketWallet} from "@tp-lab/aptos-wallet-adapter";
import {TrustWallet} from "@trustwallet/aptos-wallet-adapter";
import {WelldoneWallet} from "@welldone-studio/aptos-wallet-adapter";

import { createWeb3Modal } from '@web3modal/wagmi/react'
import { defaultWagmiConfig } from '@web3modal/wagmi/react/config'

import { WagmiProvider } from 'wagmi'
import { arbitrum, mainnet } from 'wagmi/chains'
import { createNetworkConfig, SuiClientProvider, WalletProvider } from '@mysten/dapp-kit';
import { getFullnodeUrl } from '@mysten/sui/client';
 
// Config options for the networks you want to connect to
const { networkConfig } = createNetworkConfig({
	devnet: { url: 'https://sui.devnet.m2.movementlabs.xyz'},
});


// 1. Your WalletConnect Cloud project ID
const projectId = '47763f0426c0cb4279a4ccfae07b46bf'

const wagmiConfig = defaultWagmiConfig({
  chains: [mainnet, arbitrum],
  projectId,
  metadata: {
    name: 'Movement Faucet',
    description: 'AppKit Example',
    url: 'https://web3modal.com',
    icons: ['https://avatars.githubusercontent.com/u/37784886']
  }
});

// 2. Create wagmiConfig
const metadata = {
  name: 'Movement Faucet',
  description: 'AppKit Example',
  url: 'https://web3modal.com', // origin must match your domain & subdomain
  icons: ['https://avatars.githubusercontent.com/u/37784886']
}

const chains = [mainnet, arbitrum] as const
const config = defaultWagmiConfig({
  chains,
  projectId,
  metadata,
})


import * as Sentry from "@sentry/react";
// import {BrowserTracing} from "@sentry/tracing";

import ReactGA from "react-ga4";
// import {initGTM} from "./api/hooks/useGoogleTagManager";
import {GTMEvents} from "./dataConstants";

// initGTM({
//   events: {
//     walletConnection: GTMEvents.WALLET_CONNECTION,
//     searchStats: GTMEvents.SEARCH_STATS,
//   },
// });

ReactGA.initialize(process.env.GA_TRACKING_ID || "G-8XH7V50XK7");

Sentry.init({
  dsn: "https://531160c88f78483491d129c02be9f774@o1162451.ingest.sentry.io/6249755",
  // integrations: [new BrowserTracing()],
  environment: process.env.NODE_ENV,
  enabled: process.env.NODE_ENV == "production",

  // Set tracesSampleRate to 1.0 to capture 100%
  // of transactions for performance monitoring.
  // We recommend adjusting this value in production
  tracesSampleRate: 0.5,
});

createWeb3Modal({
  wagmiConfig: config,
  projectId,
})

// inform the compiler of the existence of the window.aptos API
declare global {
  interface Window {
    aptos: any;
  }
}

const queryClient = new QueryClient();

const wallets = [
  new PetraWallet(),
  new PontemWallet(),
  new MartianWallet(),
  new FewchaWallet(),
  new RiseWallet(),
  // new MSafeWalletAdapter(),
  new NightlyWallet(),
  new OpenBlockWallet(),
  new TokenPocketWallet(),
  new TrustWallet(),
  new WelldoneWallet(),
  // Blocto supports Testnet/Mainnet for now.
  new BloctoWallet({
    network: NetworkName.Testnet,
    bloctoAppId: "6d85f56e-5f2e-46cd-b5f2-5cf9695b4d46",
  }),
];

ReactDOM.render(
  <React.StrictMode>
    <div>Hello, World!</div>
  </React.StrictMode>,
  document.getElementById("root")
);

ReactDOM.render(
  <React.StrictMode>
    <StatsigProvider
      sdkKey={
        process.env.REACT_APP_STATSIG_SDK_KEY ||
        "client-gQ2Zhz3hNYRf6CSVaczkQcZfK0yUBv5ln42yCDzTwbr"
      }
      waitForInitialization={true}
      options={{
        environment: {tier: process.env.NODE_ENV},
      }}
      user={{}}
    >
      <QueryClientProvider client={queryClient}>
        <QueryClientProvider client={queryClient}>
        <AptosWalletAdapterProvider plugins={wallets} autoConnect={true}>
        <SuiClientProvider networks={networkConfig} defaultNetwork="devnet">
        <WalletProvider>
        <WagmiProvider config={wagmiConfig}>
          <BrowserRouter>
            <Routes>
              <Route path="/" element={<LandingPage />} />
            </Routes>
          </BrowserRouter>
          </WagmiProvider>
          </WalletProvider>
          </SuiClientProvider>
          </AptosWalletAdapterProvider>
          </QueryClientProvider>
      </QueryClientProvider>
    </StatsigProvider>
  </React.StrictMode>,
  document.getElementById("root"),
);
```

**Review by andygolay — CHANGES_REQUESTED** [review]

**Review by andygolay — COMMENTED** [review]
@0xPrimata I'm not sure that I understand how it works to have users connect their wallets to get coins from the faucet, because they will need to pay gas... but if they don't have any coins already, then they can't pay gas. 

Is there some sponsored transaction mechanism in the code for Aptos Move and MEVM, or could you explain how it is supposed to work for users who have an empty wallet? I tried minting Aptos Move unsuccessfully and realized when I got the Sui transaction to prompt the wallet for signing, there may be an issue about sponsoring transactions on all three networks?

When I configure to Suzuka on Razor wallet and try to mint I get the simulation error "Cannot use 'in' operator to search for 'message' in not found", regardless of which coin I attempt to mint. Are you able to mint coins on Suzuka?

image

**Comment by 0xPrimata** [comment]
@andygolay it's not meant to be sponsored. If they need gas, they need to request native tokens right above.

**Comment by andygolay** [comment]
> @andygolay it's not meant to be sponsored. If they need gas, they need to request native tokens right above.

Oh I see. Having mint on the faucet page will likely confuse users so it should be organized as separate tabs or separate pages.

**Comment by andygolay** [comment]
@0xPrimata I fixed the errors that were thrown in the `suiMint` function as follows:

```
  async function suiMint() {

    const tokenMint = `${PACKAGE_ID}::${token.toLowerCase()}::${token}`
    const transaction = new Transaction();
    transaction.moveCall({
      target: PACKAGE_ID,
      module: 'tokens',
      function: 'mint',
      typeArguments: [tokenMint],
      arguments: []
    })

    await signAndExecuteTransaction(
      {
        transaction,
        chain: 'sui:m2',
      },
      {
        onSuccess: (result) => {
          console.log('executed transaction', result);
          setDigest(result.digest);
        },
      },
    );
  }
  ```
  
  (Changed `package` to `target` and separated the transaction creation step from the step of adding move calls which changes it to a `TransactionResult`)
  
 I'm still getting a VM error in the UI when I call it. I suspect it may be that there are args needed or something, but I need to see the deployed package's code to understand.
 
image

When I go to https://explorer.sui.devnet.m2.movementlabs.xyz/ and search for the package `0x457abead7283c8af79b0902e71decf173f88624fe8dd2e76be97b6132c39e9c9`, nothing comes up. 

image

Maybe M2 is down. https://explorer.polymedia.app/ is also failing to return transactions for M2, assuming the endpoint https://sui.devnet.m2.movementlabs.xyz:443 is still correct?

Yeah I think maybe the network is down at the moment but not being reported on the status page. Razor wallet faucet just failed for me despite it working about an hour ago.

Anyway, check the code above for the Sui mint fix [👍](https://emojipedia.org/thumbs-up)

**Comment by andygolay** [comment]
@0xPrimata I verified that M2 is up.

I'd like to check the Sui Move code in the package deployed at `0x457abead7283c8af79b0902e71decf173f88624fe8dd2e76be97b6132c39e9c9` but without explorers working, I'm not sure how to do that with explorers failing to load anything. 

Where can I find a link to the Move code? If you add it to the repo that could be most helpful for visibility.

**Comment by andygolay** [comment]
Okay I figured out how to get the M2 devnet explorer to show transactions.

The package at `0x457abead7283c8af79b0902e71decf173f88624fe8dd2e76be97b6132c39e9c9` does not contain a module named `tokens`.

image

Your code indicates a module named `tokens` and a function `mint`:

```
    transaction.moveCall({
      target: PACKAGE_ID,
      module: 'tokens',
      function: 'mint',
      typeArguments: [tokenMint],
      arguments: []
    })
```    

I don't see any `mint` functions in the modules for this packages. I'm assuming you'll want to dynamically pass in which tokens to mint as well.

Did you mean to refer to a different package ID? If you add a `move` directory to the code and include the packages you're deploying, that would be helpful.

**Review by andygolay — APPROVED** [review]
lgtm

**Pull request merged**